### PR TITLE
gui-wm/hyprland: add 0.40.0

### PR DESCRIPTION
**Pull request by MrRoy**
gui-wm/hyprland: add 0.40.0

N.B.: the reason I added both `dev-util/hyprwayland-scanner` `0.3.4` and `0.3.7` is because `gui-wm/hyprland-0.40.0` depends on `=dev-util/hyprwayland-scanner-0.3.4` (does *not* compile on `0.3.7`)
I included `0.3.7` too since I had already committed it and it will be required at some point anyway.

---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: gui-wm/hyprland/hyprland-0.40.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	virtual/pkgconfig
+"
+RDEPEND="
+	${HYPRPM_RDEPEND}
+	${WLROOTS_RDEPEND}
+	dev-cpp/tomlplusplus
+	dev-libs/glib:2
+	dev-libs/libinput
+	>=dev-libs/wayland-1.20.0
+	gui-libs/hyprcursor
```
Comment (dlan17) [inline gui-wm/hyprland/hyprland-0.40.0.ebuild]: the hyprland-0.39.1 has a dep of >=gui-libs/hyprcursor-0.1.7, can you double check any changes in this version 0.40.0?

File: gui-wm/hyprland/hyprland-0.40.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	>=dev-libs/hyprland-protocols-0.2
+	>=dev-libs/hyprlang-0.3.2
+	>=dev-libs/wayland-protocols-1.34
+"
+BDEPEND="
+	${WLROOTS_BDEPEND}
+	|| ( >=sys-devel/gcc-13:* >=sys-devel/clang-16:* )
+	app-misc/jq
+	dev-build/cmake
+	=dev-util/hyprwayland-scanner-0.3.4
```
Comment (dlan17) [inline gui-wm/hyprland/hyprland-0.40.0.ebuild:94]: sounds not really necessary, you can add 0.3.7 when at it, but anyway I'm fine with this..

File: gui-wm/hyprland/hyprland-0.40.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	fi
+}
+
+src_prepare() {
+	default
+	cd "${S}/subprojects/wlroots-hyprland" || die
+	# apply.sh script is broken in the targetted commit of 0.40.0
+	# they fixed it since; the fix being this patch
+	eapply "${FILESDIR}/wlroots-hyprland-apply-0.40.0.patch"
+}
```
Comment (dlan17) [inline gui-wm/hyprland/hyprland-0.40.0.ebuild:118]: why bother using customized src_prepare()+eapply, just make the patch apply on top of source dir?
adjust the patch like this:
+--- a/subprojects/wlroots-hyprland/patches/apply.sh
++++ b/subprojects/wlroots-hyprland/patches/apply.sh

File: gui-wm/hyprland/hyprland-0.40.0.ebuild (hunk 2 above)
Comment (MrRoy) [reply gui-wm/hyprland/hyprland-0.40.0.ebuild:94]: This is necessary. Hyprland does *not* compile with > 0.3.4

File: gui-wm/hyprland/hyprland-0.40.0.ebuild (hunk 1 above)
Comment (MrRoy) [reply gui-wm/hyprland/hyprland-0.40.0.ebuild]: Fixed now

File: gui-wm/hyprland/hyprland-0.40.0.ebuild (hunk 3 above)
Comment (MrRoy) [reply gui-wm/hyprland/hyprland-0.40.0.ebuild]: Good idea, fixed

**Pull request closed without merging**